com.unity.ml-agents/Runtime/Academy.cs

```diff
@@ -384,6 +387,8 @@ public void EnvironmentStep()
                 ForcedFullReset();
             }
 
+            AgentIncrementStep?.Invoke();
```

Comment:
Hmm, for on-demand decision, this might still have the same problem of an Agent being marked to request a decision, and that getting cleared if the Agent hits maxSteps.

I forget, is there a reason we can't do this last?

Reply:
I see, this would happen if the user does not use the decision requester. I will make some experiments to determine if we can do this last as it would make most sense.